src/configfile/Parser_helpers.cpp

```diff
@@ -0,0 +1,592 @@
+ * @param detailed toggles between detailed and simple output. (0 for simple, 1 for detailed)
+ * @param layer_num the current layer number.
+ */
+void ConfigFileParser::printLocations(std::vector<struct location>& locations, int layer, int detailed,
+                                      std::vector<int> layer_num) {
+  std::string c = "";
+  for (int i = 0; i < layer; i++) {
+    c += "   ";
+    c += "|";
+  }
+  layer_num.push_back(1);
+  for (unsigned long j = 0; j < locations.size(); j++) {
+    std::cout << "--------------------------" << std::endl;
+    std::cout << c << "Location: ";
+    for (unsigned long k = 0; k < layer_num.size(); k++) {
+      std::cout << layer_num[k];
+      if (k != layer_num.size() - 1) std::cout << ".";
+    }
+    std::cout << " name: " << locations[j].name << std::endl;
+    if (detailed) {
+      if (locations[j].root != "") std::cout << c << "Root: " << locations[j].root << std::endl;
+      if (locations[j].autoindex)
+        std::cout << c << "Autoindex: on" << std::endl;
+      else
+        std::cout << c << "Autoindex: off" << std::endl;
+      if (locations[j].methods.size() > 0) {
+        std::cout << c << "Methods: ";
+        for (unsigned long k = 0; k < locations[j].methods.size(); k++) {
+          std::cout << locations[j].methods[k] << " ";
+        }
+        std::cout << std::endl;
+      }
+      if (locations[j].return_code != 0)
+        std::cout << c << "Redirect: " << locations[j].return_code << " " << locations[j].return_url << std::endl;
+      if (locations[j].index.size() > 0) {
+        std::cout << c << "Index: ";
+        for (unsigned long k = 0; k < locations[j].index.size(); k++) {
+          std::cout << locations[j].index[k] << " ";
+        }
+        std::cout << std::endl;
+      }
+    }
+    // go one step deeper
+    if (locations[j].locations.size() > 0) {
+      // layer_num.push_back(1);
+      printLocations(locations[j].locations, layer + 1, detailed, layer_num);
+    }
+    layer_num.back()++;
+  }
+}
+
+int ConfigFileParser::testFunction(std::string const& key, std::vector<std::string>& args, int& line_count) {
+  int i;
+  if (key == "server")
+    i = server(args, line_count, 0);
+  else if (key == "listen")
+    i = listen(args, line_count, 1);
+  else if (key == "server_name")
+    i = serverName(args, line_count, 1);
+  else if (key == "error_page")
+    i = errorPage(args, line_count, 1);
+  else if (key == "client_max_body_size")
+    i = clientMaxBodySize(args, line_count, 1);
+  else if (key == "location")
+    i = location(args, line_count, 1);
+  else if (key == "limit_except")
+    i = limitExcept(args, line_count, 2);
+  else if (key == "return")
+    i = returnKeyword(args, line_count, 2);
+  else if (key == "root")
+    i = root(args, line_count, 2);
+  else if (key == "autoindex")
+    i = autoindex(args, line_count, 2);
+  else if (key == "index")
+    i = index(args, line_count, 2);
+  else
+    i = 2;
+  return i;
+}
```

Comment:
This solution seems like a work around to test pvt member functions. I am sure there is a better way